internal/users/idgenerator.go

```diff
@@ -133,15 +134,15 @@ func getIDCandidate(minID, maxID uint32, usedIDs []uint32) (uint32, error) {
 	}
 
 	// Try IDs above the highest used
-	for id := highestUsed + 1; id <= maxID; id++ {
+	for id := highestUsed + 1; id <= maxID && highestUsed < math.MaxUint32; id++ {
 		if _, found := slices.BinarySearch(usedIDs, id); found {
 			continue
 		}
 		return id, nil
 	}
 
 	// Fallback: try IDs from minID up to highestUsed
-	for id := minID; id <= highestUsed && id <= maxID; id++ {
+	for id := minID; id <= highestUsed && id <= maxID && id < math.MaxUint32; id++ {
```

Comment:
Mmh, this check has the effect that we will never try the ID `math.MaxUint32`, even if it's a valid ID. How about this instead:

```go
	for id := minID; id <= highestUsed && id <= maxID; id++ {
		if _, found := slices.BinarySearch(usedIDs, id); !found {
			return id, nil
		}
		if id == math.MaxUint32 {
			break // Avoid overflow
		}
	}
```

Reply:
See my proposal in https://github.com/ubuntu/authd/pull/993#discussion_r2185044185

Reply:
> Mmh, this check has the effect that we will never try the ID `math.MaxUint32`, even if it's a valid ID

It did not, this loop, but it was already picked by the other loop if that was the case (there's a test checking it), given that if the max is free to use it's also the latest we've iterated to.